introducing rstudio addin for dso repro stage

DSchreyer opened the pull request:
## v0.5.1

### New Features

-   Introduced the rstudio addin to `dso repro` the current stage

DSchreyer commented:
local testing required

tschwarzl commented on R/addin.R:
```diff
@@ -0,0 +1,49 @@
+    dvc_yaml_path <- file.path(stage_path, "dvc.yaml")
+
+    tryCatch({
+      message(glue::glue("Reproducing the current stage"))
```
Add to "#' @importFrom glue glue" for roxygen to resolve import dependencies

tschwarzl commented on R/addin.R:
```diff
@@ -0,0 +1,49 @@
+      message(glue::glue("Report generated: {report_file}"))
+      message(glue::glue("Displaying report file: {report_file}"))
+      # Check the file extension and display in the viewer
+          rstudioapi::viewer(report_file)
```
import / importFrom with rstudioapi

grst replied:
technically not necessary when using `::`. 
Afaik the pre-commit checks also ensure that one does not forget to add it to DESCRIPTION

tschwarzl reviewed (commented):
Very nice. 

I would like to like discuss an option to add
"dso repro stage" - with dependencies
"dso repro stage -s" - only single stage
"dso repro stage -f" - force.

grst commented on CHANGELOG.md:
```diff
@@ -4,6 +4,16 @@ All notable changes to this project will be documented in this file.
 
+## v0.5.1
+
+### New Features
```
Basically whenever you add a new feature, it should bump the `minor` version, so in this case v0.6.0.

grst commented on R/addin.R:
```diff
@@ -0,0 +1,51 @@
+      result <- system2(DSO_EXEC,
+                        c("repro -s", shQuote(dvc_yaml_path)))
+      if (result != 0) {
+        stop("System command failed with status: {result}")
```
`glue` seems to be missing here

DSchreyer commented:
```Pre-commit hooks are not installed in this project.
This hooks will take care of running consistency checks and automatically 
syncing dvc.
Do you want to install the pre-commit hooks now? [y/n]: Aborted.```

pre commit hooks lead to error which is not identified and can not be resolved within rstudio and the current addin

grst commented:
So do I get this right that not the actual hooks are the problem, but dso asking whether to install the hooks?

DSchreyer commented:
> So do I get this right that not the actual hooks are the problem, but dso asking whether to install the hooks?

correct. DSO asks to installl but the addin is not interactive and therefore it results in an error. When the hooks are installed or dso doesnt ask then it works normally

grst commented:
ok, then I propose adding an ENV var or command line flag to the dso CLI to not ask that can be used from R

DSchreyer commented:
[11/07/24 10:10:47] INFO     Compiling a total of 29 config files.              
[11/07/24 10:10:49] INFO     Configuration compiled successfully.               
                    INFO     Running `dvc repro -s                <stagename>/dvc.yaml/dvc.yaml`    
                             
             
Another issue arose: recursive dvc.yaml addition to path

DSchreyer commented:
Fixed the issue named above and removed the show report function. The show report function froze the session when it was too large. Report can be opened manually inside Rstudio

grst reviewed (approved):
just two nitpicky things left; LGTM
